scvi/model/base/_log_likelihood.py

```diff
@@ -13,22 +14,34 @@ def compute_elbo(vae, data_loader, feed_labels=True, **kwargs):
     It still gives good insights on the modeling of the data, and is fast to compute.
     """
     # Iterate once over the data and compute the elbo
-    elbo = 0
+    if return_mean:
+        elbo = 0
+    else:
+        elbo = np.array([])
     for tensors in data_loader:
         _, _, scvi_loss = vae(tensors, **kwargs)
 
-        recon_loss = scvi_loss.reconstruction_loss_sum
-        kl_local = scvi_loss.kl_local_sum
-        elbo += (recon_loss + kl_local).item()
+        recon_loss = np.sum(
+            [np.array(i) for i in scvi_loss.reconstruction_loss.values()], axis=0
+        )
+        kl_local = np.sum([np.array(i) for i in scvi_loss.kl_local.values()], axis=0)
 
-    kl_global = scvi_loss.kl_global_sum
+        if return_mean:
+            elbo += (recon_loss + kl_local).sum(0).item()
+        else:
+            elbo = np.concatenate((elbo, recon_loss + kl_local), axis=0)
+
+    kl_global = np.sum([np.array(i) for i in scvi_loss.kl_global.values()], axis=0)
     n_samples = len(data_loader.indices)
-    elbo += kl_global
-    return elbo / n_samples
+    if return_mean:
+        elbo += kl_global
+        return elbo / n_samples
+    else:
+        return elbo + kl_global / n_samples
```

Comment:
Probably best to also use torch operations here since `compute_reconstruction_error` doesn't use numpy. Let me know what you think

Reply:
Fine with me. Can you puh those changes?
```suggestion
        return elbo + kl_global / n_samples
```